types/src/contract_event.rs

```diff
@@ -156,10 +158,8 @@ impl ContractEvent {
     }
 
     pub fn is_new_epoch_event(&self) -> bool {
-        match self {
-            ContractEvent::V1(event) => *event.key() == new_epoch_event_key(),
-            ContractEvent::V2(_event) => false,
-        }
+        self.type_tag() == NEW_EPOCH_EVENT_MOVE_TYPE_TAG.deref()
```

Comment:
Makes sense on the state sync side 😄 i.e., assuming this definition is complete and we're not missing any edge cases, it should be sufficient for the event subscription service to notify components of the epoch change.